src/base/memory_monitor_sr.cpp

```diff
@@ -393,49 +538,41 @@ namespace cubperf
 /* APIs */
 int mmon_initialize (const char *server_name)
 {
+  int error = NO_ERROR;
   cubperf::mmon_Gl = new cubperf::memory_monitor (server_name);
 
   if (cubperf::mmon_Gl == nullptr)
     {
-      /* error case */
-      return -1;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (cubperf::memory_monitor));
```

Comment:
Need to define the error codes for the memory monitor module first. And use them.